### ISSUE TITLE
Tensorflow shape detection fails with multiple versions.

### ISSUE THREAD

**Issue by nieksand** [NVIDIA/tensorrt-inference-server#432]
**Issue**: Automatic shape detection for TensorFlow models seems to break when there is more that one version in the model repository.  (Details below).

**Expected Behavior**: I would expect TRTIS to verify that all versions have the same shape and to then use that shape.

I can reproduce on the 19.04, 19.05, and 19.06 releases.

---

My config.pbtxt is minimal:
```
platform: "tensorflow_savedmodel"
version_policy: { all { } }
```

Everything is fine with just one version:
```
prompt> find 1/
1/
1/model.savedmodel
1/model.savedmodel/saved_model.pb
```

But if I make a second version, even with the exact same model, shape detection no longer works...
```
prompt> cp -R 1/ 2/

prompt> docker run --rm --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 -p8000:8000 -v /data/cape/models_dev/repository/:/models/ -it nvcr.io/nvidia/tensorrtserver:19.06-py3 trtserver --model-store=/models/ --strict-model-config=false
```

Error from TRTIS logs:
```
E0703 14:41:47.017690 1 model_repository_manager.cc:761] failed to load 'rcr_v3' version 2: Invalid argument: unable to load model 'rcr_v3', configuration expects 0 inputs, model provides 1
E0703 14:41:47.017870 1 model_repository_manager.cc:761] failed to load 'rcr_v3' version 1: Invalid argument: unable to load model 'rcr_v3', configuration expects 0 inputs, model provides 1
```

Status then shows `MODEL_UNAVAILABLE`.

If I explicitly set the shape in config.pbtxt, both versions load fine:
```
platform: "tensorflow_savedmodel"
version_policy: { all { }}
input {
  name: "input"
  data_type: TYPE_FP32
  dims: -1
  dims: 224
  dims: 224
  dims: 3
}
output {
  name: "output"
  data_type: TYPE_FP32
  dims: -1
  dims: 8
}
```

**Comment by CoderHam** [comment]
I was able to reproduce this error. Will look into it.

**Comment by CoderHam** [comment]
If you check your log you should see. (use --log-verbose 1)
```TensorFlow SavedModel autofill: Internal: unable to autofill for '<your_savedmodel_name>' due to multiple versions```
The autofill is currently conservative and allows for only 1 model version. We will relax this check later in the future. For now your solution is to use only 1 model version if you need to use the autofill feature. 
If this does not resolve your issue you may re-open the issue.

**Comment by CoderHam** [comment]
Fixed by #436 
You should now be able to autofill with multiple versions. Keep in mind that for now the _first_ version will be used for autofill.

**Comment by nieksand** [comment]
Excellent.  Thank you!
